bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComRFXSensorMessage.java

```diff
@@ -0,0 +1,204 @@
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.rfxcom.internal.exceptions.RFXComException;
+import org.openhab.binding.rfxcom.internal.exceptions.RFXComUnsupportedChannelException;
+import org.openhab.binding.rfxcom.internal.exceptions.RFXComUnsupportedValueException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigDecimal;
+
+public class RFXComRFXSensorMessage extends RFXComDeviceMessageImpl<RFXComRFXSensorMessage.SubType> {
+    private static final BigDecimal ONE_HUNDRED = BigDecimal.valueOf(100);
+
+    private final Logger logger = LoggerFactory.getLogger(RFXComRFXSensorMessage.class);
+
+    public enum SubType implements ByteEnumWrapper {
+        TEMPERATURE(0),
+        A_D(1),
+        VOLTAGE(2),
+        MESSAGE(3);
+
+        private final int subType;
+
+        SubType(int subType) {
+            this.subType = subType;
+        }
+
+        @Override
+        public byte toByte() {
+            return (byte) subType;
+        }
+    }
+
+    public SubType subType;
+
+    public int sensorId;
+    private Double temperature;
+    private BigDecimal miliVoltageTimesTen;
+    public byte signalLevel;
+
+    public RFXComRFXSensorMessage() {
+        super(PacketType.RFXSENSOR);
+    }
+
+    public RFXComRFXSensorMessage(byte[] data) throws RFXComException {
+        encodeMessage(data);
+    }
+
+    @Override
+    public String toString() {
+        String str = super.toString();
+
+        str += ", Sub type = " + subType;
+        str += ", Device Id = " + getDeviceId();
+        str += ", Temperature = " + temperature;
+        str += ", Voltage = " + getVoltage();
+        str += ", Signal level = " + signalLevel;
+
+        return str;
+    }
+
+    @Override
+    public void encodeMessage(byte[] data) throws RFXComException {
+        super.encodeMessage(data);
+
+        subType = fromByte(SubType.class, super.subType);
+
+        sensorId = (data[4] & 0xFF);
+
+        byte msg1 = data[5];
+        byte msg2 = data[6];
+
+        if (subType == SubType.TEMPERATURE) {
+            encodeTemperatureMessage(msg1, msg2);
+        } else if (subType == SubType.A_D) {
+            encodeVoltageMessage(msg1, msg2);
+        } else if (subType == SubType.VOLTAGE) {
+            encodeVoltageMessage(msg1, msg2);
+        } else if (subType == SubType.MESSAGE) {
+            encodeStatusMessage(msg2);
+        }
+
+        signalLevel = (byte) ((data[7] & 0xF0) >> 4);
+    }
+
+    private void encodeTemperatureMessage(byte msg1, byte msg2) {
+        temperature = (short) ((msg1 & 0x7F) << 8 | (msg2 & 0xFF)) * 0.01;
+        if ((msg1 & 0x80) != 0) {
+            temperature = -temperature;
+        }
+    }
+
+    private void encodeVoltageMessage(byte msg1, byte msg2) {
+        miliVoltageTimesTen = BigDecimal.valueOf((short) ((msg1 & 0xFF) << 8 | (msg2 & 0xFF)));
+    }
+
+    private void encodeStatusMessage(byte msg2) {
+        //logger
```

Comment:
There is a TODO, what should be done and can it be completed before merging the PR?